data_steward/cdr_cleaner/cleaning_rules/deid/generate_site_mappings_and_ext_tables.py

```diff
@@ -0,0 +1,164 @@
+)
+""")
+
+
+class GenerateSiteMappingsAndExtTables(BaseCleaningRule):
+    """
+    Generates non-deterministic site_ids for hpo sites.
+    """
+
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id,
+                 mapping_dataset_id):
+        """
+        Initialize the class with proper info.
+
+        Set the issue numbers, description and affected datasets.  As other
+        tickets may affect this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = f'Change PIDs to RIDs in specified tables'
+
+        self.mapping_dataset_id = mapping_dataset_id
+
+        super().__init__(issue_numbers=ISSUE_NUMBERS,
+                         description=desc,
+                         affected_datasets=[cdr_consts.CONTROLLED_TIER_DEID],
+                         project_id=project_id,
+                         dataset_id=dataset_id,
+                         sandbox_dataset_id=sandbox_dataset_id,
+                         affected_tables=[])
+
+    def get_query_specs(self):
+        """
+        Return a list of dictionary query specifications.
+
+        :return:  A list of dictionaries.  Each dictionary contains a
+            single query and a specification for how to execute that query.
+            The specifications are optional but the query is required.
+        """
+        query_list = []
+        query = dict()
+        query[cdr_consts.QUERY] = SITE_MAPPINGS_QUERY.render(
+            project_id=self.project_id,
+            dataset_id=self.sandbox_dataset_id,
+            table_id=SITE_TABLE_ID,
+            site_prefix=EHR_SITE_PREFIX,
+            lookup_tabels_dataset=LOOKUP_TABLES_DATASET_ID,
+            hpo_site_id_mappings_table=HPO_SITE_ID_MAPPINGS_TABLE_ID,
+            ppi_pm=PPI_PM,
+            rdr=RDR)
+        query_list.append(query)
+
+        # gather queries to generate ext tables
+        query_list.append(
+            get_generate_ext_table_queries(self.project_id,
+                                           self.sandbox_dataset_id,
+                                           self.sandbox_dataset_id,
+                                           self.mapping_dataset_id))
+        return [query]
+
+    def get_sandbox_tablenames(self):
+        return []
+
+    def setup_rule(self, client):
+        """
+        Function to run any data upload options before executing a query.
+        """
+        pass
+
+    def setup_validation(self, client):
+        """
+        Run required steps for validation setup
+        """
+        raise NotImplementedError("Please fix me.")
+
+    def validate_rule(self, client):
+        """
+        Validates the cleaning rule which deletes or updates the data from the tables
+        """
+        raise NotImplementedError("Please fix me.")
+
+
+if __name__ == '__main__':
+    import cdr_cleaner.args_parser as parser
+    import cdr_cleaner.clean_cdr_engine as clean_engine
+
```

Comment:
Perhaps wrap up this as a separate function, in addition, the same block of code is used in `data_steward/tools/generate_ext_tables.py`